runelite-client/src/main/java/net/runelite/client/ui/overlay/components/MinimapOrb.java

```diff
@@ -0,0 +1,197 @@
+	private static final Color HIGHLIGHT_COLOR = new Color(255, 255, 255, 50);
+
+	// black background
+	private final BufferedImage background;
+	// Color and reflection highlight
+	private final BufferedImage fill;
+	// Shadow for when enabled
+	private final BufferedImage fillEnabled;
+	// Highlight for recharge
+	private final BufferedImage highlight;
+	// Frame
+	private final BufferedImage frame;
+	// Icon to draw ontop
+	@Setter
+	private BufferedImage overlay;
+
+	public MinimapOrb(Color color, Color enabledColor, BufferedImage overlay)
+	{
+		{
+			background = new BufferedImage(ORB_W, ORB_W, BufferedImage.TYPE_INT_ARGB);
+			Graphics2D g = background.createGraphics();
+
+			g.setColor(BACKGROUND_COLOR);
+			g.fillOval(0, 0, ORB_W, ORB_W);
+		}
+
+		fill = createFill(color, false);
+		fillEnabled = createFill(enabledColor, true);
+
+		{
+			highlight = new BufferedImage(ORB_W, ORB_W, BufferedImage.TYPE_INT_ARGB);
+			Graphics2D g = highlight.createGraphics();
+
+			g.setColor(HIGHLIGHT_COLOR);
+			g.fillOval(0, 0, ORB_W, ORB_W);
+		}
+
+		try
+		{
+			frame = ImageIO.read(getClass().getResourceAsStream("minimap_orb_background.png"));
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+
+		this.overlay = overlay;
+	}
+
+	private BufferedImage createFill(Color color, boolean enabled)
+	{
+		BufferedImage bufferedImage = new BufferedImage(ORB_W, ORB_W, BufferedImage.TYPE_INT_ARGB);
+		Graphics2D graphics = bufferedImage.createGraphics();
+		float[] dist = {0.2f, 1f};
+
+		int r = color.getRed();
+		int g = color.getGreen();
+		int b = color.getBlue();
+		int a = color.getAlpha();
+
+		float darkenMultiplier = 0.4f;
+		Point2D center = new Point2D.Float(ORB_W / 2.7f, ORB_W / 2.7f);
+
+		if (enabled)
+		{
+			// Darken and move center to make it look like the orb is pressed down
+			darkenMultiplier = 0.20f;
+			center = new Point2D.Float((ORB_W / 1.8f), (ORB_W / 1.8f));
+			dist = new float[]{0.7f, 1f};
+		}
+
+		// Darken the base color to create a shadow effect on the edges of the orb
+		r *= darkenMultiplier;
+		g *= darkenMultiplier;
+		b *= darkenMultiplier;
+
+		Color[] colors = {color, new Color(r, g, b, a)};
+
+		// Divided by 2.7f for a gradient in the top left of the orb
+		graphics.setPaint(new RadialGradientPaint(center, ORB_W / 2.0f, dist, colors));
+
+		graphics.fillOval(0, 0, ORB_W, ORB_W);
+
+		return bufferedImage;
+	}
+
+	/**
+	 * Draws a minimap orb like the prayer/HP/run orbs.
+	 *
+	 * @param graphics              graphics to draw to
+	 * @param pos                   top left position of the orb
+	 * @param percentFilled         how far filled the orb should be
+	 * @param rechargePercentFilled how far through the recharge this stat is
+	 * @param amount                number to display on the orb
+	 * @param enabled
+	 */
+	public void render(Graphics2D graphics, Point pos, double percentFilled, double rechargePercentFilled, int amount, boolean enabled)
```

Comment:
Well when you moved this to components, this should definitely extend RenderableEntity and all these fields should be setters on the orb.

Reply:
You mean args should becomefields right?

Reply:
Yeah, so you would be able to implement the RenderableEntity and use it's render method.

Reply:
Should it then render at (0,0) like the rest of the components?

Reply:
probably should have position field like other high-level components (Text, Panel etc)